experimental/dds/tree2/src/feature-libraries/untypedTree.ts

```diff
@@ -298,4 +298,58 @@ export interface EditableTreeEvents {
 	 * @returns a visitor to traverse the subtree or `void`.
 	 */
 	subtreeChanging(upPath: UpPath): PathVisitor | void;
+
+	/**
+	 * Raised on a node right before a change is applied to one of its fields or the fields of a descendant node.
+	 *
+	 * @param upPath - The path (upward) corresponding to the location of the node raising the event. Note that this is not
+	 * necessarily the node that is about to change; the change could be in a descendant of the node the `upPath` points to.
+	 *
+	 * @remarks
+	 * What exactly qualifies as a change that triggers this event (or {@link afterChange}) is dependent on the
+	 * implementation of SharedTree. In general, these events will fire once for every atomic editing operation supported
+	 * by SharedTree; {@link beforeChange} before the change is applied, and {@link afterChange} after it is.
+	 *
+	 * {@link Sequence} fields present two exceptions:
+	 *
+	 * The first one is that events will fire separately for each node involved in the operation (when inserting, removing,
+	 * or moving more than one node at a time). This means that, for example, when inserting two nodes into a {@link Sequence}
+	 * field the following will happen:
+	 *   - {@link beforeChange} will fire once before either new node is present in the tree.
+	 *   - {@link afterChange} will fire once after the first node is present in the tree, but the second one isn't.
+	 *   - {@link beforeChange} will fire once before the second node is present in the tree, but the first one already is.
+	 *   - {@link afterChange} will fire once after the second node is present in the tree (so at this point both nodes are).
+	 * Something similar applies to removing nodes from a sequence, and moving them to another sequence.
+	 *
+	 * The second one is that for an operation to move nodes, events will fire *twice* for each node being moved; once
+	 * while they are being detached from their source location, and once they are being attached at the target location.
+	 */
```

Comment:
Interested in thoughts about these docs, and the fact that operations on sequences currently fire the events for each node individually. I'm clear that we're ok firing events twice when moving node in sequence fields, once for detaching and once for attaching. But is it also expected that they fire once per node during insertions and removals in a sequence? (And twice for moves).

Reply:
It is expected but will be fixed before release. It's listed under the "Status" section in the root README.

Docs look good to me.